### PR TITLE
DCS Outbox Publisher (GSI-772)

### PR DESCRIPTION
**Pull request by TheByronHimes**
Adds the outbox publisher for the NonStagedFileRequested event, as well as the `publish-events [--all]` command.

### CODE REVIEW

File: services/dcs/src/dcs/adapters/outbound/daopub.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        ),
+        examples=["file_downloads"],
+    )
+    unstaged_download_collection: str = Field(
+        default=...,
+        description=(
+            "The type used for event indicating that a download was requested"
+            + " for a file that is not yet available in the outbox."
+        ),
+        examples=["unstaged_download_requested"],
```
Comment (mephenor) [inline services/dcs/src/dcs/adapters/outbound/daopub.py:42]: Not sure if this has to be hyphenated instead

File: services/dcs/tests_dcs/test_outbox_dao.py [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+    async with set_new_correlation_id():
+        await joint_fixture.nonstaged_file_requested_dao.republish()
+
+        # consume the republished events with the dummy translator
+        translator = DummySubTranslator(config=joint_fixture.config)
+        async with KafkaOutboxSubscriber.construct(
+            config=joint_fixture.config, translators=[translator]
+        ) as subscriber:
+            await subscriber.run(forever=False)
+            await subscriber.run(forever=False)
```
Comment (mephenor) [inline services/dcs/tests_dcs/test_outbox_dao.py:158]: As in the other PR: wouldn't this just consume the initial insert events?

Comment (TheByronHimes) [reply services/dcs/tests_dcs/test_outbox_dao.py:161]: You're right, I misunderstood what happened in the expect_events cm. I've updated it to clear the topics right after that so the only events consumed are the republished ones.

File: services/dcs/src/dcs/adapters/outbound/daopub.py (hunk 1 above)
Comment (TheByronHimes) [reply services/dcs/src/dcs/adapters/outbound/daopub.py]: It does, I've updated the description.

**Review by mephenor — APPROVED** [review]

**Pull request merged**